dask/dataframe/io/parquet/core.py

```diff
@@ -95,6 +95,64 @@ def __call__(self, part):
         )
 
 
+ParquetFunctionWrapper = ReadParquetFunctionWrapper
+
+
+class ToParquetFunctionWrapper:
+    """
+    Parquet Function-Wrapper Class
+
+    Writes a DataFrame partition into a distinct
+    parquet file. When called, the function also
+    requires a ``partition_info`` kwarg.
+    """
+
+    def __init__(
+        self,
+        engine,
+        path,
+        fs,
+        partition_on,
+        write_metadata_file,
+        i_offset,
+        name_function,
+        kwargs_pass,
+    ):
+        self.engine = engine
+        self.path = path
+        self.fs = fs
+        self.partition_on = partition_on
+        self.write_metadata_file = write_metadata_file
+        self.i_offset = i_offset
+        self.name_function = name_function
+        self.kwargs_pass = kwargs_pass
+
+    def __call__(self, df, partition_info=None):
```

Comment:
I like this use of #8310! However, the implementation of that is not ideal: we generate all the `partition_info` dicts up front, which is kinda overkill for large numbers of partitions.

For this case, could you use a `BlockwiseDep` instead that just returns the current block index? Something like https://github.com/dask/dask/blob/a154dae7f9893965a95ecd3dde6ec0fa6868764c/dask/blockwise.py#L279-L288

With https://github.com/dask/dask/pull/7417, it seems like we're going with the pattern of having composable BlockwiseDep helpers as proposed in https://github.com/dask/dask/issues/7513. I _think_ (unconfirmed) you can even just pass them into `map_partitions` as arguments, but if not that would be a good change to make.

Reply:
Good call - I added a `BlockIndex(BlockwiseDep)` class (similar to your suggestion here).